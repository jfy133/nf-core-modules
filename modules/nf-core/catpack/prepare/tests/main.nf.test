// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test catpack/prepare
nextflow_process {

    name "Test Process CATPACK_PREPARE"
    script "../main.nf"
    process "CATPACK_PREPARE"

    tag "modules"
    tag "modules_nfcore"
    tag "catpack"
    tag "catpack/prepare"

    test("metagenome -sarscov2 - fasta") {

        when {
            process {
                """
                input[0] = Channel.of([[id:'db'], file(params.modules_testdata_base_path + '/genomics/prokaryotes/metagenome/fasta/sarscov2.fasta', checkIfExists: true)])
                input[1] = file(params.modules_testdata_base_path + 'genomics/prokaryotes/metagenome/taxonomy/taxdmp/names.dmp', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path + 'genomics/prokaryotes/metagenome/taxonomy/taxdmp/nodes.dmp', checkIfExists: true)
                input[3] = file(params.modules_testdata_base_path + 'genomics/prokaryotes/metagenome/taxonomy/accession2taxid/nucl_gb.accession2taxid', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    test("metagenome -sarscov2 - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([[id:'db'], file(params.modules_testdata_base_path + '/genomics/prokaryotes/metagenome/fasta/sarscov2.fasta', checkIfExists: true)]).mix(GUNZIP.out.gunzip).groupTuple()
                input[1] = file(params.modules_testdata_base_path + 'genomics/prokaryotes/metagenome/taxonomy/taxdmp/names.dmp', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path + 'genomics/prokaryotes/metagenome/taxonomy/taxdmp/nodes.dmp', checkIfExists: true)
                input[3] = file(params.modules_testdata_base_path + 'genomics/prokaryotes/metagenome/taxonomy/accession2taxid/nucl_gb.accession2taxid', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
