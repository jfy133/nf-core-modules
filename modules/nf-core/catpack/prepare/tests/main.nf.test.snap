{
    "metagenome -sarscov2 - fasta": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "db": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-01-15T14:43:34.68418275"
    },
    "metagenome -sarscov2 - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "db"
                        },
                        [
                            
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,d851e296d4025a8060b6283ad3b63937"
                ],
                "db": [
                    [
                        {
                            "id": "db"
                        },
                        [
                            
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d851e296d4025a8060b6283ad3b63937"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-01-15T14:43:40.694491364"
    }
}